### PR TITLE
Feature/clean audio state

### PR DESCRIPTION
**Pull request by TroyNeubauer**

### CODE REVIEW

File: src/dial.rs [hunk 1]
```diff
@@ -80,18 +80,10 @@ pub struct DialAlarm {
     pub dial_id: usize,
     pub time: Instant,
     pub correct_key: char,
+    pub sound_id: u64,
```
Comment (newcomb-luke) [inline src/dial.rs]: Good idea, I'm not sure why I didn't just do this. The only issue is that nothing uses it, it still uses the get_id() method which could be removed and replaced with just an access to this field.

File: src/dial.rs [hunk 2]
```diff
@@ -197,16 +186,19 @@ impl Dial {
...
+            self.alarm_fired = true;
 
-            Some(dial_alarm)
+            Some(DialAlarm::from(&*self))
         } else {
             None
         }
     }
 
+    pub fn play_alarm(&self, audio: &crate::AudioManager) {
```
Comment (newcomb-luke) [inline src/dial.rs]: Should this actually be a separate function? We could just pass in the AudioManager as another parameter to update()

**Review by newcomb-luke — CHANGES_REQUESTED** [review]
Nice changes to AudioManager, just 2 things should be resolved

File: src/dial.rs (hunk 1 above)
Comment (TroyNeubauer) [reply src/dial.rs]: Will do, looks like the best option is to store it like you said

File: src/dial.rs (hunk 2 above)
Comment (TroyNeubauer) [reply src/dial.rs]: Yeah passing audio to `Dial::update` is much cleaner

**Review by newcomb-luke — APPROVED** [review]
Looks good, nice cleanup

**Pull request merged**